renesas: rz-cmn: Add support for eSD card IPL booting

ThanhNguyenRVC opened the pull request:
Changes include:

- Added eSD card support in I/O storage.
- Added Makefile rule to create separate BL2 binaries.

tiennguyenuh commented on plat/renesas/rz/common/rz_common.mk:
```diff
@@ -194,3 +194,14 @@ ifneq (${TRUSTED_BOARD_BOOT},0)
+.PHONY: bl2-xspi.bin bl2-emmc.bin bl2-esd.bin
+
+define PLAT_BL2_VARIANT_RULE
+bl2-$(1).bin:
```
I think bl2-${1} is enough, no need .bin here. For example

```
make PLAT=cmn BOARD=rz_cmn bl31 bl2-emmc
make PLAT=cmn BOARD=rz_cmn bl31 bl2-esd
make PLAT=cmn BOARD=rz_cmn bl31 bl2-xspi
```

ThanhNguyenRVC replied:
It is updated

tiennguyenuh commented on plat/renesas/rz/common/rz_common.mk:
```diff
@@ -194,3 +194,14 @@ ifneq (${TRUSTED_BOARD_BOOT},0)
+	$(call SHELL_COPY,${BUILD_PLAT}/bl2.bin,${BUILD_PLAT}/bl2-$(1).bin)
+endef
+
+$(foreach variant,xspi emmc esd,$(eval $(call PLAT_BL2_VARIANT_RULE,$(variant))))
```
Is it possible to create a bl2-all rule which builds for all method?

```
make PLAT=cmn BOARD=rz_cmn bl31 bl2-all
```

ThanhNguyenRVC replied:
It is supported

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/esd.h:
```diff
@@ -0,0 +1,524 @@
+ *********************************************************************************************************************/
+/*******************************************************************************
+* System Name  : SDHI Driver
+* File Name    : sd.h
```
filename -> esd.h

ThanhNguyenRVC replied:
It is updated

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/esd_main.c:
```diff
@@ -0,0 +1,4051 @@
+ */
+
+/**********************************************************************************************************************
+ * File Name    : sd_main.c
```
File name -> esd_main.c

ThanhNguyenRVC replied:
It is updated

tiennguyenuh commented on plat/renesas/rz/common/drivers/sd/sdif.h:
Do we need this sdif.h file? I thought we're using esdif.h

ThanhNguyenRVC replied:
It is removed

ThanhNguyenRVC committed:
renesas: rz-cmn: Add support for eSD card IPL booting

Changes include:

- Added eSD card support in I/O storage.
- Added Makefile rule to create separate BL2 binaries.

Signed-off-by: Tien Nguyen <tien.nguyen.uh@renesas.com>
Signed-off-by: Thanh Nguyen <thanh.nguyen.wx@bp.renesas.com>

tiennguyenuh reviewed (approved):
Looks good to me.
